src/System.IO.Compression/tests/Performance/Perf.DeflateStream.cs

```diff
@@ -10,75 +10,70 @@ namespace System.IO.Compression.Tests
+            switch (type)
+            {    
+                case CompressionType.CryptoRandom:
+                    bytes = new byte[fileSize];
+                    var rand = RandomNumberGenerator.Create();
+                    rand.GetBytes(bytes);
+                    using (FileStream output = File.Create(filePath))
+                    using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
+                        zip.Write(bytes, 0, bytes.Length);
+                    break;
+                case CompressionType.RepeatedSegments:
+                    bytes = Text.Encoding.UTF8.GetBytes(utils.CreateString(fileSize / 1000));
+                    using (FileStream output = File.Create(filePath))
+                    using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
+                        for (int i = 0; i < 1000; i++)
+                            zip.Write(bytes, 0, bytes.Length);
+                    break;
+                case CompressionType.NormalData:
+                    bytes = Text.Encoding.UTF8.GetBytes(utils.CreateString(fileSize));
+                    using (FileStream output = File.Create(filePath))
+                    using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
                         zip.Write(bytes, 0, bytes.Length);
-                _compressedFiles.Add(new object[] { filePath });
+                    break;
             }
-            return _compressedFiles;
+            return filePath;
         }
 
-        // Creates byte arrays that contain random data to be compressed
-        public static IEnumerable<object[]> ByteArraysToCompress()
+        private static byte[] CreateBytesToCompress(CompressionType type)
         {
-            if (_byteArraysToCompress == null)
+            byte[] bytes = null;
+            PerfUtils utils = new PerfUtils();
+            switch (type)
             {
-                PerfUtils utils = new PerfUtils();
-                _byteArraysToCompress = new List<object[]>();
-
-                // Regular, semi well formed data
-                _byteArraysToCompress.Add(new object[] { Text.Encoding.UTF8.GetBytes(utils.CreateString(100000000)) });
-
-                // Crypto random data
-                {
-                    byte[] bytes = new byte[100000000];
-                    var rand = RandomNumberGenerator.Create();
-                    rand.GetBytes(bytes);
-                    _byteArraysToCompress.Add(new object[] { bytes });
-                }
-
-                // Highly repeated data
-                {
-                    byte[] bytes = new byte[101000000];
+                case CompressionType.CryptoRandom:
+                    bytes = new byte[100000000];
+                    RandomNumberGenerator.Create().GetBytes(bytes);
+                    break;
+                case CompressionType.RepeatedSegments:
+                    bytes = new byte[101000000];
```

Comment:
I messed up the char->byte conversion and bandaided the problem instead of resolving it. Fixed.